contracts/concordium-governance-committee-election/tests/tests.rs

```diff
@@ -166,14 +172,218 @@ fn test_init_config() {
     let init = initialize(&module_ref, &init_param, &mut chain).expect("Init contract succeeds");
     let invocation =
         view_config(&mut chain, &init.contract_address).expect("Can invoke config entrypoint");
-    let config: ElectionConfig = invocation.parse_return_value().expect("Can parse value");
+    let config: InitParameter = invocation.parse_return_value().expect("Can parse value");
     assert_eq!(config.admin_account, ALICE);
 }
 
+#[test]
+fn test_receive_guardian_public_key() {
+    let (mut chain, contract_address) = new_chain_and_contract();
+    let config: InitParameter = view_config(&mut chain, &contract_address)
+        .expect("Can invoke config entrypoint")
+        .parse_return_value()
+        .expect("Can parse value");
+
+    let param = vec![0, 1, 2, 5, 1, 6, 7];
+    let param_other = vec![1, 2, 3, 4, 5, 1, 2, 3];
+    register_guardian_public_key_update(&mut chain, &contract_address, &BOB_ADDR, &param)
+        .expect("Key registration should succeed");
+
+    register_guardian_public_key_update(&mut chain, &contract_address, &DANIEL_ADDR, &param_other)
+        .expect("Key registration should succeed");
+
+    register_guardian_public_key_update(&mut chain, &contract_address, &ALICE_ADDR, &param)
+        .expect_err("Key registration should fail due to not being in the list of guardians");
+
+    let contract_sender = Address::Contract(ContractAddress {
+        index:    0,
+        subindex: 0,
+    });
+    register_guardian_public_key_update(&mut chain, &contract_address, &contract_sender, &param)
+        .expect_err("Cannot register keys from contract sender");
```

Comment:
Would it be possible to check the exact error here instead of just "there is an error".  You have very precise errors for the different failure cases in the contract.

That would make the test more robust.